### ISSUE TITLE
FTBS with filesystem lib

### ISSUE THREAD

**Issue by vincentchabannes** [feelpp/feelpp#1841]
We have this error when use for example clang7 compiler and the stdlib of gcc84 : 
```
Linking CXX shared library libfeelpp_fmi4cpp.so
/usr/bin/ld: cannot find -lc++fs
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
_deps/fmi4cpp-build/src/CMakeFiles/fmi4cpp.dir/build.make:310: recipe for target '_deps/fmi4cpp-build/src/libfeelpp_fmi4cpp.so' failed
make[2]: *** [_deps/fmi4cpp-build/src/libfeelpp_fmi4cpp.so] Error 1
```

We should review the usage of filesystem lib. Maybe we can say that no need to link by default (as this lib has been removed) but and use cmake_cxx_link_flags if really need

**Comment by metivett** [comment]
Yes, this special filesystem linkage is really painful if you want to maintain retrocompatibility... I tried to address some specific combinations in https://github.com/feelpp/FMI4cpp/commit/5a6cb30edd95c0820424219008bc063b49f30cda, but it seems I did not deal properly with the case you mention (clang+gcc_stdlib). 
Maybe we could just add this case directly (https://github.com/feelpp/FMI4cpp/blob/5a6cb30edd95c0820424219008bc063b49f30cda/src/CMakeLists.txt#L123) ?

**Comment by vincentchabannes** [comment]
the issue is that we don't have an access direclty to the id of c++ standard lib with cmake. There are some tricks for that but I think that in major cases now we didn't need to link to a filesystem lib. And if really necessary, cmake_cxx_link_flags should be enough (or add a special cmake arg)

**Comment by metivett** [comment]
Or maybe we could directly test for `filesytem` linkage (one of 3 cases: no additional lib/c++fs/stdc++fs) by using check_cxx_source_compiles with a small snippet ?

**Comment by prudhomm** [comment]
@vincentchabannes @metivett isn't this issue solved now ?

**Comment by prudhomm** [comment]
Resolved in develop: commit c59f16c08f pins FMI4cpp f08a851, which removes c++fs from the Clang link path. Current supported Ubuntu/Debian and Spack configurations no longer exercise the failing linkage, and Feelpp PKG CI run 32505398509 passed the Noble package, Docker, and Apptainer builds: https://github.com/feelpp/feelpp/actions/runs/32505398509. Closing as completed.
